extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildAggregatorFactory.java

```diff
@@ -71,6 +74,7 @@ protected byte getCacheTypeId()
   public Aggregator factorize(final ColumnSelectorFactory columnSelectorFactory)
   {
     final ColumnValueSelector<Object> selector = columnSelectorFactory.makeColumnValueSelector(getFieldName());
+    validateInputs(columnSelectorFactory.getColumnCapabilities(getFieldName()));
```

Comment:
`factorizeBuffer` and `factorizeVector` should also validate their inputs

also please add tests for this stuff